plugins/redis/redis.cpp

```diff
@@ -373,35 +382,53 @@ void Connection::readReplyAndAssertWithCmdMsg(Reply * reply, const char * msg, c
     readReply(reply);
     assertOnErrorWithCmdMsg(reply->query(), msg, key);
 }
-Connection * Connection::createConnection(ICodeContext * ctx,  Connection * & _cachedConnection, const char * options, int _database, const char * password, unsigned _timeout)
+Connection * Connection::createConnection(ICodeContext * ctx,  Connection * & _cachedConnection, const char * options, int _database, const char * password, unsigned _timeout, bool cacheConnections)
 {
-#if HIREDIS_VERSION_OK
-    if (!_cachedConnection)
+#if HIREDIS_VERSION_OK_FOR_CACHING
+    //Fetch connection caching level
+    if (!connectionCachingLevelChecked)//test to guard against unnecessary critical section
```

Comment:
The code is fine, but it adds unnecessary complication to a function that is not time critical.